lib/phoenix_live_view/test/client_proxy.ex

```diff
@@ -956,6 +959,26 @@ defmodule Phoenix.LiveViewTest.ClientProxy do
     end
   end
 
+  defp maybe_js_event("[" <> _ = encoded_js) do
+    js = encoded_js |> DOM.parse() |> Phoenix.json_library().decode!()
+    op = Enum.filter(js, fn [kind, _args] -> kind == "push" end)
+
+    case op do
+      [["push", %{"event" => event} = args] | rest] ->
+        if rest != [] do
+          raise ArgumentError,
+                "the elixir Phoenix.LiveViewTest client currently only supports a single push"
+        end
+
+        {event, args["value"] || %{}}
+
+      nil ->
+        raise ArgumentError, "no event found within JS command: #{inspect(js)}"
```

Comment:
`nil` won't ever be matched and you can probably remove the nesting:

```suggestion
      [] ->
        raise ArgumentError, "no push command found within JS commands: #{inspect(js)}"

      [["push", %{"event" => event} = args]] ->
        {event, args["value"] || %{}}

      [_ | _] ->
        raise ArgumentError,
                "Phoenix.LiveViewTest currently only supports a single push within JS commands"
```